fields.go

```diff
@@ -235,19 +235,11 @@ func makeStructFields(root reflect.Type) (structFields, *SemanticError) {
 	// or the one that is uniquely tagged with a JSON name.
 	// Otherwise, no dominant field exists for the set.
 	flattened := allFields[:0]
-	slices.SortFunc(allFields, func(x, y structField) int {
-		switch {
-		case x.name != y.name:
-			return strings.Compare(x.name, y.name)
-		case len(x.index) != len(y.index):
-			return cmp.Compare(len(x.index), len(y.index))
-		case x.hasName && !y.hasName:
-			return -1
-		case !x.hasName && y.hasName:
-			return +1
-		default:
-			return 0 // TODO(https://go.dev/issue/61643): Compare bools better.
-		}
+	slices.SortStableFunc(allFields, func(x, y structField) int {
+		return cmp.Or(
```

Comment:
Sorting multiple dimensions has now become my favorite use of `cmp.Or`

Reply:
Wow never thought of this

Reply:
in this particular case it doesn't seem to be a big deal, but isn't the use of cmp.Or causing all arguments to all be computed upfront? in the old implementation, if e.g. the names didn't match, then you didn't have to run `cmp.Compare` on the index lengths.

again in this case I doubt it's a noticeable difference, but if you have a cmp.Or with five chained string comparisons, I bet it can matter if often it's the first comparison that would make the other ones unnecessary.

Reply:
This only happens once per type and then the result is cached. I'll take readability over performance in this case.